Write version string to the log when starting the wallet server.

jonathanknowles opened the pull request:
# Issue Number

#350 

# Overview

Just a small PR, but one that helps to finalize #350. 

- [x] I have added a startup message to the log for the wallet server.
- [x] The startup message includes the current version string.

Commit:
Additional tests for cardano-wallet server --database <file>, checking if <file>'s are created

Commit:
Additional test for cardano-wallet-launcher --state-dir <dir>, checking if <dir> and contents are created

Anviking committed:
getTransaction has dynamic length and shouldn't use isolate

Anviking committed:
Add keyToAddress & separate concern with get/putAddress

Previously:
    `getAddress :: ByteString 33 bytes -> Address 32 bytes`
    `putAddress :: Address 32 bytes -> ByteString 33 bytes`
Now:
    `getAddress :: ByteString 33 bytes -> Address 33 bytes`
    `putAddress Address 33 bytes -> ByteString 33 bytes`
     **Implemented** `keyToAddress :: XPub 32 bytes -> Address 33 bytes`

- Implement Jörmungandr keyToAddress
- Refactor to have singleAddressFromKey in Binary module
- Ignore chaincode in jörmungandr keyToAddress & add tests

Anviking committed:
Rename Address { getAddress} to Address { unAddress }
to avoid naming collision with Binary.getAddress.

jonathanknowles committed:
Remove manually-defined `Bounded` instance.

jonathanknowles committed:
Write version string to the log when starting the wallet server.

KtorZ commented on exe/wallet/Main.hs:
```diff
@@ -415,6 +415,9 @@ execHttpBridge args _ = do
     tracer <- initTracer minLogSeverity "serve"
+    logInfo tracer $ "Wallet backend server STARTED. "
+        <> "Version "
+        <> T.pack (showVersion version)
```
But, that is a lie :grimacing: ... The backend server hasn't started _yet_. 
There's already code doing this by the way but called by a callback once the server has started (cf 10 lines below :sweat_smile:  ... ).

This would be a better place for such logging ^^^. That's a nice idea indeed to log the version.

jonathanknowles replied:
Understood! I'll move this to the other location.

jonathanknowles replied:
In effect we already have a message to indicate that the server has **started**:
https://github.com/input-output-hk/cardano-wallet/blob/master/exe/wallet/Main.hs#L427-L428

Therefore, we probably don't need another similar message.

However, it **is** useful to output the **version string**. Therefore, I've changed this PR so that it prints "starting" rather than "started".

Commit:
Rebase and code review remarks

Anviking committed:
Enable decoding of grouped/account/multisig addresses

Commit:
add DaedalusIPC test (removed by previous rebase)

Commit:
Use shouldReturn instead of shouldBe to avoid repetition

Commit:
Use temp directory for launcher tests

Commit:
Use temp directory for serve --database test

jonathanknowles committed:
Merge pull request #420 from input-output-hk/jonathanknowles/simplify-bounded-instance

Remove manually-defined `Bounded` instance for `Severity`.

KtorZ committed:
Avoid compiling and caching binaries in CI

Instead, we pulled already compiled binaries from various sources.
Somehow, Travis stopped caching binaries in CI and I couldn't figure
out what the exact problem is, but this is blocking every PR from
getting merge so, this "work-around" should work fine and not even
be "that worse"

KtorZ committed:
upload already build-in cardano-http-bridge linux binary

KtorZ committed:
Merge branch 'KtorZ/fix-travis-cache'

KtorZ committed:
better cache build dependencies in Travis

KtorZ committed:
Merge pull request #417 from input-output-hk/anviking/219/fix-address-confusion

Jörmungandr: Separate concern between keyToAddress and get/putAddress

KtorZ committed:
Merge pull request #413 from input-output-hk/piotr/testing

Additional tests checking if files exists when using --database and --state-dir options
